src/main/java/org/wildfly/security/ssl/X509CRLTrustManager.java

```diff
@@ -0,0 +1,149 @@
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wildfly.security.ssl;
+
+import static org.wildfly.common.Assert.checkMinimumParameter;
+import static org.wildfly.common.Assert.checkNotNullParam;
+
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.cert.CRL;
+import java.security.cert.CertStore;
+import java.security.cert.CertStoreParameters;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CollectionCertStoreParameters;
+import java.security.cert.PKIXBuilderParameters;
+import java.security.cert.X509CertSelector;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+
+import javax.net.ssl.CertPathTrustManagerParameters;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import org.wildfly.security._private.ElytronMessages;
+
+/**
+ * Extension to the <code>X509TrustManager</code> interface to support CRL verification.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class X509CRLTrustManager implements X509TrustManager {
+
+    private static final int DEFAULT_MAX_CERT_PATH_LENGTH = 5;
+
+    private final TrustManager[] trustManagers;
+    private final X509Certificate[] acceptedIssuers;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param trustStore a {@link KeyStore} with the trusted certificates (must not be {@code null})
+     * @param crlFile the input stream pointing to a certificate revocation list (may be {@code null})
+     * @param maxCertPath sets the value of the maximum number of non-self-issued intermediate
+     *                    certificates that may exist in a certification path. The value must be equal or greater than 1.
+     * @param acceptedIssuers an array of certificate authority certificates which are trusted for authenticating peers (may be {@code null}).
+     */
+    public X509CRLTrustManager(KeyStore trustStore, InputStream crlFile, int maxCertPath, X509Certificate[] acceptedIssuers) {
+        checkNotNullParam("trustStore", trustStore);
+        checkMinimumParameter("maxCertPath", 1, maxCertPath);
+        try {
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
```

Comment:
Actually I see the passing in of a TrustManager[] probably doesn't work but I do think the remainder of the config options should be supported.